src/main/kotlin/dev/isxander/modstitch/base/extensions/Modstitch.kt

```diff
@@ -286,7 +292,16 @@ open class ModstitchExtensionImpl @Inject constructor(
         set(value) = if (value != null) platformExtension<BaseModDevGradleExtension> { enable { mcpVersion = value } } else {}
 
     override val minecraftVersion = objects.property<String>()
-    override val javaTarget = objects.property<Int>()
+
+    override val javaVersion = objects.property<Int>().convention(minecraftVersion.map { v ->
+        // https://minecraft.wiki/w/Tutorial:Update_Java
+        ReleaseVersion.parseOrNull(v)?.let { when {
+            it >= ReleaseVersion(1, 20, 5) -> 21
+            it >= ReleaseVersion(1, 18, 0) -> 17
+            it >= ReleaseVersion(1, 17, 0) -> 16
+            else -> 8
+        }}
+    })
```

Comment:
This doesn't handle snapshot versions. 
I wonder if Loom contains some sort of utility that does just this already.

Reply:
Yep, and that's by design. Unless you want to maintain something akin to Fabric's infamous `McVersionLookup` - the sole reason we have so many versions of Fabric Loader - the best course of action is to avoid "supporting" snapshot versions. In practice, this simply means that the defaults won't be as nice as they are for other versions, and it doesn't mean you can't build against snapshots at all. You can, you'll just need to wire some things manually.

Also, no, Loom, doesn't really care about Minecraft versions, only Fabric Loader does. So, unfortunately for us, there's nothing from the Loom's toolchain we could use to normalize Minecraft versions.

Reply:
In that regard, nothing has changed - Gradle will still throw an error when we try to access `javaVersion` in `applyJavaSettings`, stating that the `javaVersion` property on the `modstitch` extension has not been set.